File: Java/programas/POO-Constructores y Destructores/src/Rectangulo.java
//Declaración de clase derivada de Figuras.
public class Rectangulo extends Figuras{
    //Constructor que inicializa valores a los atributos públicos de la clase padre por medio de "super.".
    public Rectangulo(float base, float altura){
        super.Base = base;
        super.Altura = altura;
    }

    //Función que devuelve el área del rectángulo utilizando los atributos de la clase padre.
    public float Area(){
        return super.a = (super.Base * super.Altura);
    }

    //Función que devuelve el perímetro del rectángulo utilizando los atributos de la clase padre.
    public float Perimetro(){
        return super.p = (super.Base + super.Altura) * 2;
    }

    //Procedimiento que despliega la base, altura, área y perímetro de un rectángulo.
    public void Resultado(){
        System.out.println("---Datos del rectángulo---\n");
        System.out.println("Base del rectángulo: " + super.Base);
        System.out.println("Altura del rectángulo: " + super.Altura);
        System.out.println("Área del rectángulo: " + super.a + "u^2");
        System.out.println("Perimetro del rectángulo: " + super.p + "u");
    }

    //Destructor no necesario.
}

File: C++/programas/POO-Constructores y Destructores/POO-Constructores y Destructores/Rectangulo.cpp
#include "Rectangulo.h" //Cabecera de clase Rectangulo.
#include <iostream> //Cabecera para entrada y salida de datos.

Rectangulo::Rectangulo(float base, float altura) { //Inicializaci?n de atributos protegidos por medio de constructor.
	Base = base;
	Altura = altura;
}

Rectangulo::~Rectangulo() { //Despliega un mensaje por eliminaci?n de objeto.
	std::cout << "\nSe ha eliminado el objeto Rectangulo.\n";
}

//Funci?n que rgresa el ?rea de un rect?ngulo.
float Rectangulo::Area() {
	return a = Base * Altura;
}

//Funci?n que regresa el per?metro de un rect?ngulo.
float Rectangulo::Perimetro() {
	return p = (Base + Altura) * 2;
}

//Procedimiento que despliega informaci?n de un rect?ngulo.
void Rectangulo::Resultado() {
	std::cout << "\n--Datos del rect?ngulo--\n\n";
	std::cout << "Base del rect?ngulo: " << Base << "u.\n";
	std::cout << "Altura del rect?ngulo: " << Altura << "u.\n";
	std::cout << "?rea del rect?ngulo: " << a << "u^2.\n";
	std::cout << "Per?metro del rect?ngulo: " << p << "u.\n\n";
}